### PR TITLE
Inventory Harvester QA tests aligment

### LINKED ISSUE #29097: Align Inventory Harvester integration tests with indexer templates
# Description

The current QA testing strategy for the *Inventory Harvester* component relies on a _mock_ OpenSearch template for indexing and validation. This diverges significantly from the behavior of the actual indexer environment, which enforces strict OpenSearch index templates with explicitly defined field mappings and data type constraints. As a result, using a mock template in QA fails to reveal data format inconsistencies that could lead to ingestion or indexing errors in production.

To improve test reliability and alignment with production behavior, the QA tests must be modified to leverage the same OpenSearch templates used by the indexer. 

## Objectives

The following goals define the scope of this task:

- Integrate the actual OpenSearch indexer template into the Inventory Harvester's QA test suite, replacing the current mock version.
- Guarantee that all test data generated and indexed through the QA pipeline complies with the field types and schema rules enforced by the indexer templates.
- Enforce schema validation during testing so that any deviation from the production mappings results in a test failure, accurately reflecting real-world conditions.

### Acceptance Criteria

The revised QA test must fully utilize the indexer templates, without any mocked or relaxed validations. The test must be capable of capturing any incompatibilities or schema mismatches and must fail in such cases. A successful run must ensure that all data indexed during testing adheres strictly to the production field definitions, with no indexing errors or ingestion anomalies.

#### Definition of Done

- [x] The QA test suite references and loads the templates located at `src/wazuh_modules/inventory_harvester/indexer/template`.
- [x] All tests execute successfully under the constraints imposed by the indexer templates, with no indexing or type-related errors.
- [x] Schema violations in test data are correctly detected and cause test failures, allowing for immediate debugging and resolution.

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
| Related issue   |
|----------------------|
| Closes #29097 |

# Objective

This pull request refactors the `integration` tests for the Inventory Harvester component. The primary goal is to improve maintainability, performance, and readability—**without introducing any changes to the production source code or the test tool itself**. All enhancements are strictly scoped to the test infrastructure and execution logic.

## Description

To preserve the integrity of the test tool's implementation, all adjustments were made within the test suite itself. Specifically, a **generic and minimal index template** was introduced due the testtool requeriment params, a template is required, but no specific behavior or mappings are needed.

### `dummy_template.json`

```json
{
  "index_patterns": ["wazuh-states-*"],
  "priority": 1,
  "template": {
    "settings": {},
    "mappings": {},
    "aliases": {}
  }
}
```

This template acts as a no-op placeholder, satisfying structural requirements without altering indexing behavior.

---

### Key changes

- **Centralized logging setup using `basicConfig`**  
  Ensures consistent logging across all test components, facilitating easier debugging.

- **Template discovery via `discover_templates()`**  
  Centralizes and simplifies logic for collecting valid templates, filtering out updates or normalized variants.

- **Template registration with `create_index_from_template()`**  
  Extracts index creation logic into a reusable, testable utility function that also handles known OpenSearch edge cases.

- **Graceful handling of existing indices**  
  Prevents redundant creation and avoids failures when an index already exists.

- **Logical reordering of test flow**  
  Separates environment setup, execution, and cleanup into distinct phases, improving test readability and reproducibility.

- **Formatting source code**
  The code is aligned with the `PEP` standard.
---

#### Test Output

```console
└─$ python -m pytest -x -rA wazuh_modules/inventory_harvester/qa/
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.13.2, pytest-7.2.2, pluggy-1.5.0
rootdir: /home/gvalenzuela/Documents/Work/wazuh_back/src
plugins: md-report-0.6.2, metadata-3.1.1, html-4.1.1
collected 70 items                                                                                                                                                               

wazuh_modules/inventory_harvester/qa/test_indexation.py ......................................................................                                             [100%]

===================================================================================== PASSES =====================================================================================
======================================================================== 70 passed in 1134.07s (0:18:54) =========================================================================
```

### CODE REVIEW

File: .github/workflows/4_testintegration_inventory-harvester.yml [hunk 1]
```diff
@@ -51,4 +51,5 @@ jobs:
         run: |
           cd src
           echo "{}" > states_update_mappings.json
+          echo '{"index_patterns":["wazuh-states-*"],"priority":1,"template":{"settings":{},"mappings":{},"aliases":{}}}' > dummy_template.json
```
Comment (pereyra-m) [inline .github/workflows/4_testintegration_inventory-harvester.yml:54]: This overrides all the pre-uploaded indexes.

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Please change the approach.
The current proposal replaces the templates with a generic one, so we aren't testing what we need.

You can verify this by modifying the templates, the tests always pass.

File: .github/workflows/4_testintegration_inventory-harvester.yml (hunk 1 above)
Comment (GabrielEValenzuela) [reply .github/workflows/4_testintegration_inventory-harvester.yml:54]: I change the index for this

```json
{"index_patterns":["dummy-*"],"priority":1,"template":{"settings":{},"mappings":{},"aliases":{}}}
```

And the result was correct

```console
================================================================= 70 passed in 1117.64s (0:18:37) =================================================================
```

The idea of use this dummy index is only for the requirement of the module. We can see it on today's meet

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Gj!
The last changes reveal the tests are working.
There is just one case for hotfixes that shouldn't happen.

**Review by pereyra-m — APPROVED** [review]

**Comment by MiguelazoDS** [comment]
- Before changes 

```console
root@jammy:/home/vagrant/wazuh# curl http://localhost:9200/wazuh-states-files-*/_mapping
{"wazuh-states-files-cluster01":{"mappings":{"dynamic":"true","properties":{"agent":{"properties":{"id":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"name":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"version":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}},"file":{"properties":{"gid":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"group":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"hash":{"properties":{"md5":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"sha1":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"sha256":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}},"mtime":{"type":"date"},"owner":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"path":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}},"size":{"type":"long"},"uid":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}},"wazuh":{"properties":{"schema":{"properties":{"version":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}}}}}}}}root@jammy:/home/vagrant/wazuh# 
```

- After changes 

I can see that now the setting is "strict"
```console
root@jammy:/home/vagrant/wazuh# curl http://localhost:9200/wazuh-states-files-*/_mapping
{"wazuh-states-files-cluster01":{"mappings":{"dynamic":"strict","date_detection":false,"properties":{"@timestamp":{"type":"date"},"agent":{"properties":{"host":{"properties":{"architecture":{"type":"keyword","ignore_above":1024},"ip":{"type":"ip"}}},"id":{"type":"keyword","ignore_above":1024},"name":{"type":"keyword","ignore_above":1024},"version":{"type":"keyword","ignore_above":1024}}},"file":{"properties":{"gid":{"type":"keyword","ignore_above":1024},"group":{"type":"keyword","ignore_above":1024},"hash":{"properties":{"md5":{"type":"keyword","ignore_above":1024},"sha1":{"type":"keyword","ignore_above":1024},"sha256":{"type":"keyword","ignore_above":1024}}},"inode":{"type":"keyword","ignore_above":1024},"mtime":{"type":"date"},"owner":{"type":"keyword","ignore_above":1024},"path":{"type":"keyword","ignore_above":1024,"fields":{"text":{"type":"text"}}},"size":{"type":"long"},"uid":{"type":"keyword","ignore_above":1024}}},"wazuh":{"properties":{"cluster":{"properties":{"name":{"type":"keyword","ignore_above":1024},"node":{"type":"keyword","ignore_above":1024}}},"schema":{"properties":{"version":{"type":"keyword","ignore_above":1024}}}}}}}}}
```

**Review by MiguelazoDS — APPROVED** [review]
Just some doubts, because I do not have the context of this and I don't want to delay the merge of the implementation. The goal seems to be achieved. 

- What is the dummy_template.json line in the github workflow for? I do not find any for that. 
- Shouldn't we reuse the state_update_mappings and delegate the responsibility to the mapping update to the IndexerConnector module. I see that the proper implementation is in the python script. Just asking, it's working I mentioned in the comment.
